release-notes/9.0/os-packages.json

```diff
@@ -0,0 +1,421 @@
+                    "command-parts": ["add", "{packageName}"]
+                }
+            ],
+            "releases": [
+                {
+                    "name": "Alpine 3.20",
+                    "release": "3.20",
+                    "packages": [
+                        {
+                            "id": "libgcc",
+                            "name": "libgcc"
+                        },
+                        {
+                            "id": "ca-certificates",
+                            "name": "ca-certificates"
+                        },
+                        {
+                            "id": "openssl",
+                            "name": "libssl3"
+                        },
+                        {
+                            "id": "libstdc++",
+                            "name": "libstdc++"
+                        },
+                        {
+                            "id": "icu",
+                            "name": "icu-libs"
+                        },
+                        {
+                            "id": "icu",
+                            "name": "icu-data-full"
+                        },
+                        {
+                            "id": "tzdata",
+                            "name": "tzdata"
+                        },
+                        {
+                            "id": "krb5",
+                            "name": "libgssapi-krb5-2"
+                        }
+                    ]
+                },
+                {
+                    "name": "Alpine 3.19",
+                    "release": "3.19",
+                    "packages": [
+                        {
+                            "id": "libgcc",
+                            "name": "libgcc"
+                        },
+                        {
+                            "id": "ca-certificates",
+                            "name": "ca-certificates"
+                        },
+                        {
+                            "id": "openssl",
+                            "name": "libssl3"
+                        },
+                        {
+                            "id": "libstdc++",
+                            "name": "libstdc++"
+                        },
+                        {
+                            "id": "icu",
+                            "name": "icu-libs"
+                        },
+                        {
+                            "id": "icu",
+                            "name": "icu-data-full"
+                        },
+                        {
+                            "id": "tzdata",
+                            "name": "tzdata"
+                        },
+                        {
+                            "id": "krb5",
+                            "name": "libgssapi-krb5-2"
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "Debian",
```

Comment:
The removal of Debian 11 made me realize that this content should be in sync with https://github.com/dotnet/core/blob/main/release-notes/9.0/supported-os.json. In other words, one could imagine a validation script that checks that all the distros listed here can be cross-referenced to supported-os.json. I see that supported-os.json has both an `id` and `name` for each distro. What's the need for both? Should this file reference by `id` instead?

Similarly for the release, you list a friendly name like `Debian 12 (Bookworm)`. Why is that detailed information presented here (for a file that is package-centric) when that information isn't even contained in supported-os.json? Really, I see that information as being superfluous. You just need to know the distro id/name and version. In fact, I would argue that `release` should be renamed to `version` to more clearly indicate the relationship between these two files.

Reply:
The friendly name is so that it is easy to generate markdown. Certainly, we could use the following implementation: $"{distro} {version}". I like having the codenames in the markdown since we use both the version numbers and codenames in different parts of our infra. It also enables a long distro name like "Red Hat Enterprise Linux", which doesn't fit very nicely in an "id" field.

https://github.com/dotnet/core/pull/9412/files#diff-c3805f9c7f5bc85d9d552816365e70fad4a2243ba220a607e2a1c1360e44fc17R33

In the other file, the `id` field can be thought of as `endoflifedate-id`. Those values match what endoflife.date uses: https://endoflife.date/. The tools we wrote query that database extensively. The `name` field is what is used in the generated markdown.

Excellent point on keeping in sync. I'm going to be writing some more tools related to staying in sync. That's a good scenario to include.

Reply:
I haven't published a schema for `supported-os.json` yet. I'll make sure the connection with endoflife.date is documented.